### PR TITLE
[ENH]: Shorten Sphinx references in codebase

### PR DESCRIPTION
**Pull request by synchon**
* [ ] fix #(issue number)
* [x] description of feature/fix
* [x] tests added/passed
* [x] add an entry for the latest changes

This PR shortens Sphinx references used throughout code and docs. Also, runs `black` and adds a shield for it in `README.md`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 ![PyPI - Wheel](https://img.shields.io/pypi/wheel/junifer?style=flat-square)
 ![GitHub](https://img.shields.io/github/license/juaml/junifer?style=flat-square)
 ![Codecov](https://img.shields.io/codecov/c/github/juaml/junifer?style=flat-square)
+![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg?style=flat-square)](https://github.com/psf/black)
```
Comment (fraimondo) [inline README.md:10]: There's an error here.

**Review by fraimondo — CHANGES_REQUESTED** [review]

File: docs/extending/datagrabber.rst [hunk 2]
```diff
@@ -186,29 +186,29 @@ in the yaml file to ``ExampleBIDSDataGrabber``. Remember that we still need to
...
 
 
-Optional: Using datalad 
+Optional: Using datalad
 """""""""""""""""""""""
 
-If you are using `datalad`_, you can use the 
-:py:class:`~junifer.datagrabber.PatternDataladDataGrabber` instead of the 
+If you are using `datalad`_, you can use the
+:py:class:`~junifer.datagrabber.PatternDataladDataGrabber` instead of the
```
[the remark this answers is not in the document]
Comment (synchon) [reply docs/extending/datagrabber.rst:199]: What do you mean?

[the remark this answers is not in the document]
Comment (fraimondo) [reply docs/extending/datagrabber.rst:199]: it has the full reference: ``~junifer.datagrabber.PatternDataladDataGrabber``

[the remark this answers is not in the document]
Comment (synchon) [reply docs/extending/datagrabber.rst:199]: Yeah missed that, good catch, thanks!

**Pull request merged**